textworld/generator/game.py

```diff
@@ -447,6 +447,9 @@ def deserialize(cls, data: Mapping) -> "Game":
         if "main_quest" in data:
             game.main_quest = Quest.deserialize(data["main_quest"])
 
+        # Replace default KB with the one saved with the game.
+        game.world._state._logic = game.kb.logic
```

Comment:
I think we should add `GameLogic` as a parameter to `State.deserialize()` rather than mess with private fields.

Reply:
Ideally, we wouldn't need a State object in world. We only need a set of propositions, actually.